Code/GraphMol/Bond.cpp

```diff
@@ -291,6 +291,78 @@ void Bond::setStereoAtoms(unsigned int bgnIdx, unsigned int endIdx) {
   atoms.push_back(endIdx);
 };
 
+namespace {
+const Atom *findHighestCIPNeighbor(const Atom *atom, const Atom *skipAtom) {
+  unsigned bestCipRank = 0;
+  const Atom *bestCipRankedAtom = nullptr;
+  const auto &mol = atom->getOwningMol();
+  for (ROMol::ADJ_ITER_PAIR atomIter = mol.getAtomNeighbors(atom);
+       atomIter.first != atomIter.second; ++atomIter.first) {
+    const Atom *neighbor = mol.getAtomWithIdx(*atomIter.first);
+    if (neighbor == skipAtom) {
+      continue;
+    }
+    unsigned cip = 0;
+    if (!neighbor->getPropIfPresent(common_properties::_CIPRank, cip)) {
+      // If at least one of the atoms doesn't have a CIP rank, the highest rank
+      // does not make sense, so return a nullptr.
+      return nullptr;
+    }
+    if (cip > bestCipRank || bestCipRankedAtom == nullptr) {
+      bestCipRank = cip;
+      bestCipRankedAtom = neighbor;
+    } else if (cip == bestCipRank) {
+      // This also doesn't make sense if there is a tie (if that's possible).
+      // We still keep the best CIP rank in case something better comes around
+      // (also not sure if that's possible).
+      bestCipRankedAtom = nullptr;
+    }
+  }
+  return bestCipRankedAtom;
+}
+
+INT_VECT *findStereoAtoms(const Bond *bond) {
+  if (!bond->hasOwningMol()) {
+    return new INT_VECT;
+  }
+  switch (bond->getStereo()) {
+    case Bond::BondStereo::STEREOE:
+    case Bond::BondStereo::STEREOZ: {
+      const Atom *startStereoAtom =
+          findHighestCIPNeighbor(bond->getBeginAtom(), bond->getEndAtom());
+      const Atom *endStereoAtom =
+          findHighestCIPNeighbor(bond->getEndAtom(), bond->getBeginAtom());
+
+      if (startStereoAtom != nullptr && endStereoAtom != nullptr) {
+        int startStereoAtomIdx = static_cast<int>(startStereoAtom->getIdx());
+        int endStereoAtomIdx = static_cast<int>(endStereoAtom->getIdx());
+
+        return new INT_VECT({startStereoAtomIdx, endStereoAtomIdx});
+
+        break;
+      }
+    }
+    default:
+      return new INT_VECT;
+  }
+}
+
+}  // namespace
+
+const INT_VECT &Bond::getStereoAtoms() const {
+  if (!dp_stereoAtoms) {
+    const_cast<Bond *>(this)->dp_stereoAtoms = findStereoAtoms(this);
```

Comment:
Have you seen examples where E/Z is set but the stereoAtoms aren't? It seems like the stereoAtoms should be set whenever the E/Z is set.

Reply:
Hmmm. I think you are wrong. Have a look at `setStereoAtoms()`: they are only required for CIS/TRANS stereo, and I think most of the E/Z stereo bonds don't ever bother to set them: after all, we know the bond will be referenced to the highest CIP neighbors... question is: which atoms are those? (that's what I wanted to solve)

Setting the stereoAtoms on calculation of the E/Z was the other option, and it would make sense to have it done at the time of stereo calculation, but in case we are just reading molecules from a file or converting a SMILES string, this would slow down things. I made the implementation in the getter to only get those atoms when they are actually needed (if they ever are).

Reply:
Oh, OK. That's interesting and makes sense. I guess I still don't have enough context for intuition about how some internal stuff works.

Reply:
Yeah, this is sorely under-documented. Getting a start on fixing that is on my "soon" todo list.